protostar-rust/src/cheatcodes_hint_processor.rs

```diff
@@ -377,37 +418,38 @@ fn execute_cheatcode_hint(
                 .expect("Failed to get contract_address from return_data");
             let contract_address = Felt252::from_bytes_be(contract_address.bytes());
 
-            insert_value_to_cellref!(vm, deployed_contract_address, contract_address)?;
-            // todo in case of error, consider filling the panic data instead of packing in rust
-            insert_value_to_cellref!(vm, panic_data_start, Felt252::from(0))?;
-            insert_value_to_cellref!(vm, panic_data_end, Felt252::from(0))?;
-
-            Ok(())
+            // TODO: in case of error, consider filling the panic data instead of packing in rust
+            insert_at_pointer(vm, &mut result_segment_ptr, Felt252::from(0)).unwrap();
+            insert_at_pointer(vm, &mut result_segment_ptr, contract_address).unwrap();
```

Comment:
why you're not calling `insert_value_to_cellref` right away? I don't see a gain here, is there one?

Reply:
result_segment_ptr is a pointer, is it possible to call it on pointers?

Reply:
I don't think it's possible to use that. We're inserting values to the memory segment referenced by `result_segment_ptr`, which is a Relocatable, not a `CellRef`.

Reply:
I disagree, I mean I didn't check this on my machine but statically it seems to be possible and I don't see any benefit of introducing `result_segment_ptr` in the first place. You execute a single cheat code at a time anyway. You can pass `output_start` (it's a `CellRef`) and insert to memory using that, should work just fine.

Reply:
you can pass `output_start` instead of `result_segment_ptr ` in `match_cheatcode_by_selector`

Reply:
I'm not sure I understand. With general cheatcode, the result of an execution is just a Span<Felt252>. We have to insert values at pointer, because different cheats will have different length of result. How would inserting at a single cellref work?

Reply:
yes, discussed offline,it's ok